File: BasicMaths/GreatestCommonDivisor/GCD.ts
class GCD {
  findGCD(a: number, b: number): number {
    if (b == 0) return a;
    return this.findGCD(b, a % b);
  }

  main(): void {
    const gcd = this.findGCD(9, 12);
    console.log("GCD of two numbers: " + gcd);
  }
}

const gcd = new GCD();
gcd.main();

File: BasicMaths/GreatestCommonDivisor/GCD.py
class GCD:
    def findGCD(self, a, b):
        if b == 0:
            return a
        return self.findGCD(b, a % b)
    
    def run(self):
        a = int(input("Enter the value of a: "))
        b = int(input("Enter the value of b: "))
        gcd = self.findGCD(a, b)
        print("GCD of two numbers: "+str(gcd))
    
gcd = GCD()
gcd.run()